### ISSUE TITLE
Style/MutableConstant false positive for heredoc with frozen_string_literal

### ISSUE THREAD

**Issue by denys-husiev-perfectial** [rubocop/rubocop#10070]
Rubocop asks to freeze constant string(written with multiline heredoc style) when `# frozen_string_literal: true`

--------

## Expected behavior

No error occures

## Actual behavior

[Correctable] Style/MutableConstant: Freeze mutable objects assigned to constants

## Steps to reproduce the problem

```
# frozen_string_literal: true

class MyClass

  MY_CONSTANT = <<~LUA # raises Style/MutableConstant offence
    lorem ipsum
    lorem ipsum
  LUA

end
```

## RuboCop version

```
rubocop -V
1.20.0 (using Parser 3.0.2.0, rubocop-ast 1.11.0, running on ruby 3.0.2 x86_64-darwin20)
  - rubocop-faker 1.1.0
  - rubocop-performance 1.11.5
  - rubocop-rails 2.11.3
  - rubocop-rspec 2.4.0

.rubocop.yml
  TargetRubyVersion: 3.0
```

**Comment by Zhao-Andy** [comment]
Hello, I was working on a related example I was wondering, this should be frozen after all right? I ran the Rubocop auto correcter and it corrected to this:

```ruby
# frozen_string_literal: true

class MyClass

  MY_CONSTANT = <<~LUA.freeze # calling .freeze does not raise Style/MutableConstant offense
    lorem ipsum
    lorem ipsum
  LUA

end
```

Feel free to correct me if I'm wrong as I don't work with frozen HEREDOCS too much. :)

## RuboCop version
```
rubocop -V
1.22.0 (using Parser 3.0.2.0, rubocop-ast 1.12.0, running on ruby 3.0.2 x86_64-linux)
  - rubocop-performance 1.11.5
  - rubocop-rails 2.12.2
  - rubocop-rspec 2.5.0
  
.rubocop.yml
  TargetRubyVersion: 3.0
```

**Comment by dvandersluis** [comment]
Your heredoc would be frozen, and that `.freeze` triggers an `Style/RedundantFreeze` offense:

```sh
test.rb:5:17: C: [Correctable] Style/RedundantFreeze: Do not freeze immutable objects, as freezing them has no effect.
  MY_CONSTANT = <<~LUA.freeze # calling .freeze does not raise Style/MutableConstant offense
                ^^^^^^^^^^^^^
```

What was the original code that was autocorrected to your example?

**Comment by Zhao-Andy** [comment]
Original code was:
```ruby

class MyClass

  MY_CONSTANT = <<~LUA
    lorem ipsum
    lorem ipsum
  LUA

end
```

Actually figured it out -- I think I had some cached version of Rubocop that was running (1.20.0).

Sorry for the noise, and thanks for the explanation!

**Comment by dvandersluis** [comment]
Glad you figured it out @Zhao-Andy!
